addons/mail/static/src/core_ui/message.js

```diff
@@ -139,6 +154,10 @@ export class Message extends Component {
                 $(this.messageBody.el).find(".o-mail-read-more-less").remove();
                 this.insertReadMoreLess($(this.messageBody.el));
             }
+            if (this.shadowMessageBody.el) {
+                this.shadowRoot = this.shadowMessageBody.el.attachShadow({ mode: "open" });
+                this.state.isOriginalEmailPreview = false;
```

Comment:
Can be set in `useState()`:
```js
useState({ isOriginalEmailPreview: false })
```

Reply:
I want the `useEffect` set on `isOriginalEmailPreview` to be called only after I set the shaowRoot. So, that's why I am setting it over here rather then setting it in `useState` in `setup` which will call it before the `onMounted`

Reply:
You don't necessarily have to wait for setting the shadow root to set it to false.
`isOriginalEmailPreview` is specific to `message.type === "email"`. Shadow root is an implementation details for theses messages that is unnecessary for setting this state.

Reply:
The code in this `useEffect` should be called in the `onMounted` for `message.type === 'email'`.
Currently it wastes another rendering by setting `state.isOriginalEmailPreview` in the `onMounted()`.

If you define `state.isOriginalEmailPreview` in setup and call inner-function of `useEffect`, then you don't have to trigger 2 rendering when mounting an email message.

Reply:
okay.